content/docs/migration_1.0/Rust.md

```diff
@@ -0,0 +1,380 @@
+		  	 .with_attachment(attachment.build())
+		  	 .res()
+		  	 .await
+		  	 .unwrap();
+```
+
+- Zenoh 1.0.0
+
+```rust
+// Difference 1: No AttachmentBuilder anymore
+//               Accept any type which can be transformed into ZBytes
+let mut hashmap = HashMap::new();
+hashmap.insert(String::from("key1"), String::from("value1"));
+hashmap.insert(String::from("key2"), String::from("value2"));
+let the_attachment = ZBytes::from(&hashmap);
+// Difference 2: no with_attachment()
+publisher
+    .put(payload)
+    .attachment(the_attachment)
+    .await
+    .unwrap();
+```
+
+##  API changes in Query & Queryable
+
+Query and Queryable have been slightly reworked. 
+
+For the API replying to a `Query` from a `Queryable` declared on a session: 
+The reply function has been split into 3 separate functions variants depending on the type of reply the user wants to send.
+
+- Zenoh 0.11.x
+
+```rust
+let reply_ok = Ok(Sample::new(key_expr.clone(), payload.clone())); // Success
+query.reply(reply_ok).res().await.unwrap();
+// or 
+let reply_err = Err(Value::from(payload.clone()));                 // Failure
+query.reply(reply_err).res().await.unwrap();
+```
+
+- Zenoh 1.0.0
+
+```rust
+// No need to send Result
+// For sending Succesful Reply to Query
+query.reply(key_expr.clone(), payload.clone()).await.unwrap();  // Success
+// For sending Error Reply to Query
+query.reply_err(payload.clone()).await.unwrap();                // Failure
+// For sending Delete reply to Query (Sample Kind = Delete)
+query.reply_del(payload.clone()).await.unwrap();                // Delete (Success)
+```
+
+For how a Get `Query` receives the reply: 
+use `result()` on the `Reply` to get the `&Sample`, 
+or `into_result` to take the ownership of the `Sample`.
+
+ `Ok` variant replies, will return `Sample`. 
+ `Err` variant replies, will return `ReplyError`
+
+- Zenoh 0.11.x
+
+```rust
+while let Ok(reply) = replies.recv_async().await {
+    match reply.sample {  // sample should be Result<Sample, Value>
+        Ok(sample) => println!(
+            ">> Received ('{}': '{}')",
+            sample.key_expr.as_str(),
+            sample.value,
+        ),
+        Err(value_err) => println!("{}", String::try_from(&value_err).unwrap()),
+    }
+}
+```
+
+- Zenoh 1.0.0
+
+```rust
+while let Ok(reply) = replies.recv_async().await {
+    // Difference 1: using result() to get Result<&Sample, &ReplyError>
+    match reply.result() {
+        Ok(sample) => {
+            println!(
+                ">> Received ('{}': '{}')",
+                sample.key_expr().as_str(),
+                // Difference 2: payload() instead of value
+                sample.payload().deserialize::<String>().unwrap()
+            );
+        }
+        // Difference 3: ReplyError instead of Value
+        Err(err) => {
+            println!("{}", err.payload().deserialize::<String>().unwrap());
+        }
+    }
+}
+```
+
+We have also added the ability to get underlying Handlers from `Queryables`, a
```

Comment:
It feels like the sentence was cut abruptly

Reply:
I elaborated a little bit more regarding getting the underlying Handler.